auth/callback/interface.go

```diff
@@ -0,0 +1,28 @@
+package callback
+
+import (
+	"fmt"
+)
+
+type Unsupported struct {
+	Callback Interface
+}
+
+func (uc *Unsupported) Error() string {
+	return fmt.Sprintf("Unsupported callback <%T>: %v", uc.Callback, uc.Callback)
+}
+
+type Interface interface {
```

Comment:
I thought it would add clarity when looking at code outside the callback package:
```go
handler := callback.HandlerFunc(callbacks ...callback.Interface) {
    // ...
})
```
and then someone could review the docs for callback.Interface and understand how it's used in the scheme of things. Unfortunately I never wrote docs.